src/sentry/static/sentry/app/components/splitDiff.tsx

```diff
@@ -0,0 +1,121 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from '@emotion/styled';
+import {diffChars, diffWords, diffLines, Change} from 'diff';
+
+const diffFnMap = {
+  chars: diffChars,
+  words: diffWords,
+  lines: diffLines,
+};
+
+type Props = {
+  base: string;
+  target: string;
+  type: keyof typeof diffFnMap;
+  className: string;
+};
+
+export function SplitDiff({className, type, base, target}: Props) {
+  const diffFn = diffFnMap[type];
+
+  if (typeof diffFn !== 'function') {
+    return null;
```

Comment:
can this really happen even if we have type checking + propTypes verification and know that one of these three types is function?